src/project_root.rs

```diff
@@ -24,3 +35,31 @@ pub fn find_project_root(roots: &[String], path: &str) -> String {
         }
     }
 }
+
+pub fn gather_env_roots(language: &str) -> HashSet<PathBuf> {
+    let prefix = format!("KAK_LSP_PROJECT_ROOT_{}", language.to_uppercase());
+    debug!("Searching for vars starting with {}", prefix);
+    let p = prefix.as_str();
+    let mut roots: HashSet<PathBuf> = HashSet::new();
+    env::vars()
```

Comment:
Idiomatic Rust would be like:
```
env::vars()
      .into_iter()
      .filter(|(k, _v)| k.starts_with(p))
      .map(|(_k, v)| PathBuf::from(v))
      .collect()
```
Note that intermediate variable `root` is eliminated and there is no semicolon after `collect` so it will be returned as the result of the function.

Reply:
Oh that's neat.